cmd/gapit/memory.go

```diff
@@ -0,0 +1,293 @@
+		if err != nil {
+			return log.Err(ctx, err, "Failed to load the capture")
+		}
+		verb.At = []uint64{uint64(boxedCapture.(*service.Capture).NumCommands) - 1}
+	}
+
+	boxedVal, err := client.Get(ctx, (&path.Metrics{
+		Command: capture.Command(verb.At[0], verb.At[1:]...),
+		Type:    path.Metrics_MEMORY_BREAKDOWN,
+	}).Path())
+	if err != nil {
+		return log.Errf(ctx, err, "Failed to load metrics")
+	}
+
+	mem := boxedVal.(*api.Metrics).Metrics.(*api.Metrics_MemoryBreakdown).MemoryBreakdown
+
+	allocationFlags := []*service.Constant{}
+	if mem.AllocationFlagsIndex != -1 {
+		boxedConstants, err := client.Get(ctx, (&path.ConstantSet{
+			Api:   mem.Api,
+			Index: uint32(mem.AllocationFlagsIndex),
+		}).Path())
+		if err != nil {
+			return log.Errf(ctx, err, "Failed to load allocation flag names")
+		}
+		constants := boxedConstants.(*service.ConstantSet)
+		// If not a bitfield, we can't compare it against the flags
+		if constants.IsBitfield {
+			allocationFlags = constants.Constants
+		}
+	}
+
+	fmt.Printf("%v memory allocations\n", len(mem.Allocations))
+	sort.Slice(mem.Allocations, func(i, j int) bool {
+		return mem.Allocations[i].Handle < mem.Allocations[j].Handle
+	})
+	for _, alloc := range mem.Allocations {
+		fmt.Println("Name:", alloc.Name)
+		fmt.Println("\tDevice:      ", alloc.Device)
+		fmt.Println("\tMemory Type: ", alloc.MemoryType)
+		fmt.Println("\tSize:        ", alloc.Size)
+
+		if alloc.Flags != 0 && len(allocationFlags) != 0 {
+			fmt.Println("\tFlags:")
+			for _, f := range allocationFlags {
+				if (alloc.Flags & uint32(f.Value)) != 0 {
+					fmt.Printf("\t\t%v\n", f.Name)
+				}
+			}
+		}
+
+		if alloc.Mapping.Size != 0 {
+			fmt.Printf("\tMapped into host memory at 0x%x\n",
+				alloc.Mapping.MappedAddress)
+			fmt.Println("\t\tOffset:", alloc.Mapping.Offset)
+			fmt.Println("\t\tSize:  ", alloc.Mapping.Size)
+		}
+
+		bindings := bindingSlice(alloc.Bindings)
+		sort.Slice(bindings, bindings.bindingLess)
+		fmt.Printf("\t%v bindings:\n", len(bindings))
+		for _, binding := range bindings {
+			var typ string
+			switch binding.Type.(type) {
+			case *api.MemoryBinding_Buffer:
+				typ = "Buffer"
+			case *api.MemoryBinding_Image:
+				typ = "Image"
+			case *api.MemoryBinding_SparseImageBlock:
+				typ = "Sparse Image Block"
+			case *api.MemoryBinding_SparseImageMetadata:
+				typ = "Sparse Image Metadata"
+			case *api.MemoryBinding_SparseImageMipTail:
+				typ = "Sparse Image Mip Tail"
+			case *api.MemoryBinding_SparseOpaqueImageBlock:
+				typ = "Sparse Opaque Image Block"
+			case *api.MemoryBinding_SparseBufferBlock:
+				typ = "Sparse Buffer Block"
+			}
+			fmt.Printf("\t%v: %v\n", typ, binding.Name)
+
+			fmt.Println("\t\tOffset:", binding.Offset)
+			fmt.Println("\t\tSize:  ", binding.Size)
+
+			stringifyAspects := func(aspects []api.AspectType) string {
+				if len(aspects) == 0 {
+					return ""
+				}
+				strs := make([]string, len(aspects))
+				for i, asp := range aspects {
```

Comment:
You can add a `Format()` method on the type so that this automatically handled by the `fmt` package.
There are many examples of this [here](https://github.com/google/gapid/blob/master/gapis/service/path/path.go#L156).